cdn/cdn_generator/directory_listing.py

```diff
@@ -0,0 +1,114 @@
+import os
+import pathlib
+
+from cdn_generator.helpers import list_folder
+
+
+def _generate_folders_yaml(prefix, folders):
+    if not folders:
+        return
+
+    folders_content = [
+        "folders:",
+    ]
+    for folder in folders:
+        folders_content.append(f"- name: {folder}")
+
+    pathlib.Path(f"generated/{prefix}").mkdir(parents=True, exist_ok=True)
+    with open(f"generated/{prefix}/folders.yaml", "w") as f:
+        f.write("\n".join(folders_content) + "\n")
+
+
+def _generate_index_html(prefix, folders, files):
+    html_content = [
+        "<html>",
+        f"<head><title>Index of /{prefix}</title></head>",
+        '<body bgcolor="white">',
+        f"<h1>Index of /{prefix}</h1>",
+        "<hr>",
+        "<pre>",
+    ]
+
+    if prefix != "/":
+        html_content.append('<a href="../">../</a>')
+
+    for folder in folders:
+        html_content.append(f'<a href="{folder}/">{folder}/</a>')
+
+    for id, size in files.items():
+        # These sums are already in manifest.yaml; there is no need to link to
+        # them directly, as it makes the index file hard to read.
+        if id.endswith((".md5sum", "sha1sum", "sha256sum")):
+            continue
+
+        # Align with spaces; this is done by most httpds too
+        size = " " * (70 - len(id)) + str(size)
+        html_content.append(f'<a href="{id}">{id}</a> {size}')
+
+    html_content.extend(["</pre>", "<hr>", "</body>", "</html>"])
+
+    pathlib.Path(f"generated/{prefix}").mkdir(parents=True, exist_ok=True)
+    with open(f"generated/{prefix}/index.html", "w") as f:
+        f.write("\n".join(html_content) + "\n")
+
+
+def _generate_directory_listing(prefix, files, folders, reverse_folders=False):
+    print(f"Creating directory listing for {prefix} ...")
+
+    # Use this to make sure the latest version is on top.
+    if reverse_folders:
```

Comment:
this parameter feels a bit funny to me. The function isn't responsible for ensuring that `folders` is sorted, but is responsible for reversing it if requested? Either this should sort the folders, or reversing should be done outside this function